extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -2728,8 +2731,21 @@ private void startStages() throws IOException, InterruptedException
             throw new MSQException(new InsertCannotBeEmptyFault(task.getDataSource()));
           }
 
-          final ClusterByPartitions partitionBoundaries =
-              queryKernel.getResultPartitionBoundariesForStage(shuffleStageId);
+          // Q for MSQ folks
```

Comment:
There was an earlier comment up top that this might need rework. I want to understand a bit more and have put a comment in code. This gets all my new tests + existing tests to pass, so I have kept it here but need to understand in more detail and need clarification from MSQ folks